supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesInitializationActor.scala

```diff
@@ -94,7 +96,9 @@ class JesInitializationActor(override val workflowDescriptor: BackendWorkflowDes
       val upload = () => Future(path.writeAsJson(content))
 
       workflowLogger.info(s"Creating authentication file for workflow ${workflowDescriptor.id} at \n ${path.toString}")
-      Retry.withRetry(upload, isFatal = isFatalJesException, isTransient = isTransientJesException)(context.system) map { _ => () }
+      Retry.withRetry(upload, isFatal = isFatalJesException, isTransient = isTransientJesException)(context.system) map { _ => () } recover {
+        case failure => throw new IOException("Failed to upload authentication file", failure)
```

Comment:
This got longer than I wanted, tl;dr either is ok by me.

I think it's basically just the difference between `map` and `flatMap`. Either is fine although `recoverWith` seems More Right. The `recover` thing above kinda reminds me of a pattern I used to use in scala which now I avoid like the plague, where `x` is an `Option`:

    val z = x.getOrElse(throw new Exception("foo"))

It gets the job done but as my love of typing grew it started to bug me.